x-pack/plugin/deprecation/src/main/java/org/elasticsearch/xpack/deprecation/NodeDeprecationChecks.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+package org.elasticsearch.xpack.deprecation;
+
+
+import org.elasticsearch.action.admin.cluster.node.info.NodeInfo;
+import org.elasticsearch.action.admin.cluster.node.stats.NodeStats;
+import org.elasticsearch.xpack.core.deprecation.DeprecationIssue;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * Index-specific deprecation checks
+ */
+public class NodeDeprecationChecks {
+
+    static DeprecationIssue azureRepositoryChanges(List<NodeInfo> nodeInfos, List<NodeStats> nodeStats) {
+        List<String> nodesFound = nodeInfos.stream()
+            .filter(nodeInfo ->
+                nodeInfo.getPlugins().getPluginInfos().stream()
+                    .anyMatch(pluginInfo -> "repository-azure".equals(pluginInfo.getName()))
+            ).map(nodeInfo -> nodeInfo.getNode().getName()).collect(Collectors.toList());
+        if (nodesFound.size() > 0) {
```

Comment:
Should these check to see if the deprecated settings are in use, rather than just that the plugin is present? I'm not very familiar with these plugins but it looks like you can use the new settings in 6.x so it would make sense to be more specific with the warnings.

That might be too much effort for too little gain, though.

Reply:
I thought it would be too much effort for too little gain. maybe this can be discussed further in future discussions?

Reply:
do you recommend I just remove these checks from here for now?

Reply:
No, this is fine, we can revise if necessary.